src/spacepackets/ccsds/tm_frame.py

```diff
@@ -92,48 +129,78 @@ def unpack(cls, data: bytes) -> TransferFrameSecondaryHeader:
 class TmTransferFrame:
     def __init__(
         self,
+        length: int,
         primary_header: TmFramePrimaryHeader,
         secondary_header: Optional[TransferFrameSecondaryHeader],
         data_field: bytes,
         op_ctrl_field: Optional[bytes],
         frame_error_control: Optional[bytes],
     ) -> None:
+        self.length = length
         self.primary_header = primary_header
         self.secondary_header = secondary_header
         self.data_field = data_field
         self.op_ctrl_field = op_ctrl_field
         self.frame_error_control = frame_error_control
 
+    def pack(self, data: TmTransferFrame) -> bytes:
```

Comment:
not declared like a static method, but the API looks like it was intended to be one